packages/network-controller/src/NetworkController.ts

```diff
@@ -310,26 +302,14 @@ export class NetworkController extends BaseControllerV2<
     this.state.network === 'loading' && this.lookupNetwork();
   }
 
-  /**
-   * Sets a new configuration for web3-provider-engine.
-   *
-   * TODO: Replace this wth a method.
-   *
-   * @param providerConfig - The web3-provider-engine configuration.
-   */
-  set providerConfig(providerConfig: ProviderConfig) {
-    this.internalProviderConfig = providerConfig;
+  initializeProvider() {
```

Comment:
Nit: It would be great to have a JSDoc comment for this method. Our standard lint rules require them for all methods. That rule hasn't been enabled here only because of the backlog of methods missing it.